From d28d5e119a6719b32581f31be1800fabc132b5e8 Mon Sep 17 00:00:00 2001
From: Steven <zhao.yang_1@nxp.com>
Date: Sun, 28 Jul 2024 11:51:40 +0800
Subject: [PATCH 11/19] arm64: dts: add pwm function of the LED

Needs to disable tpm3 and tpm4 node to avoid gpio setting confilcts
in imx93-11x11-frdm-aud-hat.dts/imx93-11x11-frdm-mt9m114.dts
/imx93-11x11-frdm-tianma-wvga-panel.dts.

Signed-off-by: Steven <zhao.yang_1@nxp.com>
---
 .../freescale/imx93-11x11-frdm-aud-hat.dts    |  8 +++++++
 .../freescale/imx93-11x11-frdm-mt9m114.dts    |  8 +++++++
 .../imx93-11x11-frdm-tianma-wvga-panel.dts    |  8 +++++++
 .../boot/dts/freescale/imx93-11x11-frdm.dts   | 23 +++++++++++++++++++
 4 files changed, 47 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-aud-hat.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-aud-hat.dts
index 8e6386df97c7..6bc1bb150100 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-aud-hat.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-aud-hat.dts
@@ -161,6 +161,14 @@ codec {
 	};
 };
 
+&tpm3 {
+	status = "disabled";
+};
+
+&tpm4 {
+	status = "disabled";
+};
+
 &sai3 {
 	#sound-dai-cells = <0>;
 	pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-mt9m114.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-mt9m114.dts
index 14ecfc6b4308..e2bb26ea6ea9 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-mt9m114.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-mt9m114.dts
@@ -118,6 +118,14 @@ cap_device {
 	};
 };
 
+&tpm3 {
+	status = "disabled";
+};
+
+&tpm4 {
+	status = "disabled";
+};
+
 &parallel_csi {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dts
index f3e921ae95b5..27c8ba2f0bfe 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dts
@@ -18,6 +18,14 @@ &ldb_phy {
 	status = "disabled";
 };
 
+&tpm3 {
+	status = "disabled";
+};
+
+&tpm4 {
+	status = "disabled";
+};
+
 / {
 	backlight: backlight {
 		compatible = "pwm-backlight";
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm.dts
index 4ef8e729f3f5..d219e244d53b 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm.dts
@@ -661,6 +661,18 @@ &wdog3 {
 	status = "okay";
 };
 
+&tpm3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_led1>;
+	status = "okay";
+};
+
+&tpm4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_led2>;
+	status = "okay";
+};
+
 &iomuxc {
 	pinctrl_eqos: eqosgrp {
 		fsl,pins = <
@@ -984,6 +996,17 @@ MX93_PAD_GPIO_IO10__LPSPI3_SOUT     0x3fe
 			MX93_PAD_GPIO_IO11__LPSPI3_SCK      0x3fe
 		>;
 	};
+	pinctrl_led1: ledsgrp1 {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO04__TPM3_CH0  0x02
+			MX93_PAD_GPIO_IO12__TPM3_CH2  0x02
+		>;
+	};
+	pinctrl_led2: ledsgrp2 {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO13__TPM4_CH2  0x02
+		>;
+	};
 };
 
 &epxp {
-- 
2.25.1

